std/algorithm/searching.d

```diff
@@ -1307,7 +1309,55 @@ private auto extremum(alias map = "a", alias selector = "a < b", Range,
     return extremeElement;
 }
 
-@safe pure nothrow unittest
+private auto extremum(alias selector = "a < b", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range &&
+        !is(typeof(unaryFun!selector(ElementType!(Range).init))))
```

Comment:
The problem here is that with string mixins `a[0]` can be a valid unary and binary function: 

```
 pragma(msg, is(typeof(unaryFun!`a[0]`([0])))); // true
 pragma(msg, is(typeof(binaryFun!`a[0]`([0], [0])))); // true
```